### ISSUE TITLE
[sklearn#34320] Path to ship wheels from scikit-learn-release

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#336]
**Mirrored from [scikit-learn#34320](https://github.com/scikit-learn/scikit-learn/issues/34320)**

Original title: Path to ship wheels from scikit-learn-release
Original labels: Needs Decision
Created: 2026-06-17T11:57:21Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

I created https://github.com/scikit-learn/scikit-learn-release/ to move wheel building out of this repo. Once it's all configure, this is what releasing will look like:

1. In this repo, create a new release branch and prepare it for release. Open any PRs and merge in changes.
2. Trigger the [Publish sdist and wheels](https://github.com/scikit-learn/scikit-learn-release/actions/workflows/wheels.yml) with the "chain" option. Also set `ref` to the branch name on `scikit-learn/scikit-learn`.
3. With the "chain" option, the wheels are built first and stored as artifacts. The Github UI will need a release manager to trigger the publish to `testpypi`.
    - The "chain" will publish to `testpypi` first and then `pypi` afterwards.
4. After testpypi publish succeeds, push a tag to `scikit-learn/scikit-learn` for the release.
5. In the Github UI, the release manager triggers the final publish to `pypi`.
### What is left to do

- [ ] Update [README.md](https://github.com/scikit-learn/scikit-learn-release/blob/main/README.md) with instructions
- [ ] For both `testpypi` and `pypi` environments, add "Required Reviewers" in https://github.com/scikit-learn/scikit-learn-release/settings/environments . These will be the people that can trigger the release.
- [x] Reenable CI in `scikit-learn-release` to run all tests. I have been using `pytest --pyargs sklearn.tests.test_dummy` for a faster iteration loop. [Example](https://github.com/scikit-learn/scikit-learn-release/blob/87e247e3f11b78d3cdfb72e0dbbf191b05ca4be9/tools/test_source.sh#L17)
- [x] In [wheels.yml](https://github.com/scikit-learn/scikit-learn-release/blob/main/.github/workflows/wheels.yml), add secrets for nightly uploads for `scientific-python-nightly-wheels`
- [ ] In [wheels.yml](https://github.com/scikit-learn/scikit-learn-release/blob/main/.github/workflows/wheels.yml), configure trusted publishing for test-pypi and pypi 

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to https://github.com/godofecht/flow-scikit/issues/87 in the flow-scikit repository. Further library-specific triage and implementation will happen there.
